om4labs/diags/moc/moc.py

```diff
@@ -157,17 +185,36 @@ def arguments(cliargs=None):
         "-t", "--topog", type=str, default=None, help="Path to topog file"
     )
 
-    return parser.parse_args(cliargs)
+    if template is True:
+        return parser.parse_args(None).__dict__
+    else:
+        return parser.parse_args(cliargs)
 
 
-def read(infile, basin, gridspec, topog, varname="vmo"):
+def read(dictArgs, varname="vmo"):
     """MOC plotting script"""
 
-    print("Reading input data")
+    infile = dictArgs["infile"]
+
+    if dictArgs["model"] is not None:
+        # use dataset from catalog, either from command line or default
+        cat_platform = (
+            f"catalogs/{dictArgs['model']}_catalog_{dictArgs['platform']}.yml"
+        )
+        catfile = pkgr.resource_filename("om4labs", cat_platform)
+        cat = intake.open_catalog(catfile)
```

Comment:
will fail if ressource is not available

Reply:
What's your suggestion?  Check to see if the catalog exists and fail more gracefully?

Reply:
exactly!